src/_macosx.m

```diff
@@ -267,6 +267,39 @@ - (int)index;
 
 /* ---------------------------- Python classes ---------------------------- */
```

Comment:
Not clear what these changes have to do w/ this PR...

Reply:
It defers creating the main loop until you actually create a figure.  This means you should be able to import pyplot and the use `switch_backend` as much as you want between any backend until you create a figure.